extensions/typescript-language-features/src/tsServer/nodeManager.ts

```diff
@@ -0,0 +1,130 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { TypeScriptServiceConfiguration } from '../configuration/configuration';
+import { setImmediate } from '../utils/async';
+import { Disposable } from '../utils/dispose';
+
+
+const useWorkspaceNodeStorageKey = 'typescript.useWorkspaceNode';
+
+export class NodeVersionManager extends Disposable {
+	private _currentVersion: string | undefined;
+
+	public constructor(
+		private configuration: TypeScriptServiceConfiguration,
+		private readonly workspaceState: vscode.Memento
+	) {
+		super();
+
+		this._currentVersion = this.configuration.globalNodePath || undefined;
+		if (vscode.workspace.isTrusted) {
+			if (this.configuration.localNodePath) {
+				if (this.useWorkspaceNodeSetting === undefined) {
+					setImmediate(() => {
+						this.promptAndSetWorkspaceNode();
+					});
+				}
+				else if (this.useWorkspaceNodeSetting) {
+					this._currentVersion = this.configuration.localNodePath;
+				}
```

Comment:
They're the same except the version below uses `this.updateActiveVersion` instead of assigning directly as in `this._currentVersion = this.configuration.localNodePath` (which is kinda special because we're initializing the class at that point), so didn't seem worth it to extract into a method with an option